.docker/bin/dsh

```diff
@@ -0,0 +1,453 @@
+			upfind "docker-compose.yml" yml_path
+			cd $yml_path
+			docker-compose up -d
+		fi
+	elif is_linux && is_docker_compose ; then
+		echo "Restarting containers..."
+		upfind "docker-compose.yml" yml_path
+		cd $yml_path
+		docker-compose up -d
+	fi	
+}
+
+# output status of boot2docker if stopped or containers if started
+status ()
+{
+	is_vm_stopped
+	stopped=$?
+	case $stopped in
+		2)
+			return
+			;;
+		1)
+			if is_docker_compose ; then
+				upfind "docker-compose.yml" yml_path
+				cd $yml_path
+				docker-compose ps
+			fi
+			;;
+		0)
+			echo "boot2docker is not running"
+			if ! is_vm_exist ; then
+				return
+			else
+				echo "Run \"dsh up\" to start boot2docker"
+			fi
+			;;
+	esac
+}
+
+# start interactive bash in container
+_bash ()
+{
+	upfind "docker-compose.yml" yml_path
+	cd $yml_path
+	if is_docker && is_docker_compose ; then
+		docker exec -it $(docker-compose ps -q web) bash -i
+		RES=$?
+		if [ $RES -ne 0 ]; then
+			echo
+			echo "dsh: You might want to run \"dsh up\""
+		fi
+	fi
+}
+
+# Run a command in container in docroot
+# param $* command with it's params to run
+run ()
+{
+	if is_docker && is_docker_compose ; then
+		upfind "docker-compose.yml" yml_path
+		cd $yml_path
+		docker exec -it $(docker-compose ps -q web) bash -ic "$*"
+	fi
+}
+
+# Run a command in container changing dir to the same folder
+# param $* command with it's params to run
+_run ()
+{
+	if is_docker && is_docker_compose ; then
+		local cd_path=''
+		cmdpath cd_path
+		run "cd $cd_path && $*" 
+	fi
+}
+
+# start interactive mysql shell
+mysql ()
+{
+	if ! get_drush_path; then return; fi
+	local sql_connect
+	get_mysql_connect sql_connect
+	_run $sql_connect" -A" # -A option to speed up mysql load
+}
+
+# Truncate db and import from sql dump
+# @param $1 filename of backup file. Should be inside docroot
+mysql_import ()
+{
+	if ! get_drush_path; then return; fi
+	local docroot
+	get_docroot_path docroot
+	local filename=`basename $1`
+	local filepath
+	get_abs_path $1 filepath
+	#substract docroot path from filepath
+	local pathdiff=${filepath#$docroot}
+
+	if [[ $pathdiff == $filepath ]] ; then
+		# substraction did not succeed because file is outside docroot
+		echo "\"$filename\" should be inside your docroot folder"
+		return
+	fi
+
+	#run "ls $filename"
+
+	local sql_connect
+	get_mysql_connect sql_connect
+	#_run 'echo "use drupal; show tables;" | '$sql_connect
+	echo "Importing $filename into database 'drupal'..."
+	run $sql_connect" < $filename"
+}
+
+# Shortcut to run drush cc ... on site
+# @param $1 type of cache to clean (all, css-js, ...)
+clear_cache ()
+{
+	if ! get_drush_path; then return; fi
+	local type='all'
+	if [[ ! $1 == '' ]]; then 
+		type=$1
+	fi
+	_run 'drush cc '$type
+}
+
+# command bindings
+argv=("$@")
+case $1 in
+	up)
+		up
+		;;
+	start)
+		up
+		;;
+	stop)
+		down
+		;;
+	down)
+		down
+		;;
+	halt)
+		down
+		;;
+	restart)
+		restart
+		;;
+	reload)
+		restart
+		;;
+	status)
+		status
+		;;
+	st)
```

Comment:
I suggest we stick with `ps` and not `st` (`status`) here, to match the underlying command being used, which is `docker ps`